### DISCUSSION TITLE
Tower Defense - Development Update

### DISCUSSION THREAD

**Discussion by Freaky2164** [Freaky2164/TowerDefense#42] [General]
Hi all,

We wanted to give u a short update about our current development status. In the last weeks we tried to implement some new functionalities and made some changes to our current implementation. 

First of all we created new waypoints. Our old implementation was based on a lot of checkpoints, where each enemy was moving to one by one. Once an enemy reached a checkpoint, the enemy moved to the next one, until it got destroyed or reached the end of the map. This made the movement of our mushrooms very clunky. If you had a look at our [Risk Management](https://github.com/argastle/TowerDefense/blob/main/Projektmanagement/Risk%20Management/Risk%20Management.png) table, we are always aware of new functionalities from unity. We noticed that on a newer version of unity, they added updates regarding waypoints. To explain it in a basic way, we are now able to create waypoints that work like a path or a line. Our enemies now walk directly on a specific path, which made the movement way smoother, especially on corners. 

Also, we added music and sounds to our game. We decided to use some free sounds we could find online. Of course, some people don't wanna play with music, so we created our game settings. The settings are stored on the local device in a JSON format. Each time you start the game, your settings will be loaded. 

We wanted to change our menu design already a long time ago. Last week we finally had the time to do so. In the picture, u can see our updated menu. 

![image](https://user-images.githubusercontent.com/97874901/235479120-369a7329-c42b-46d3-b81b-5ab1faf74170.png)

We always work on new turrets. Until now, we included only our basic tower. We are currently in development, including a tower that deals AOE damage. This means that the projectiles explode on hitting an enemy and deals damage to multiple enemies. We finished the code implementation, but aren't completely finished with the design. So we will show you this Tower in future posts.

For our next tasks we want to add functions like upgrading towers, more maps, more towers or upgrades outside a round. 

We will give you another update as far as we implemented some of those tasks.

Greetings
TD Team

**Comment by erika2fr** [comment]
Hello Team Tower Defence,

It's really intressting how you develop your game. It´s a good idea to include some music to your game. In my opinion it´s a nice add on that makes your project very unique.

The menue design looks very nice. Are there some implementations that improve the user experience, e.g. a hover effect?

Furthermore, good luck with the development.

Team Common

**Comment by KevStr06** [comment]
Hi TD Team, 

Great to hear about the progress and updates to your game! The implementation of new waypoints sounds like a significant improvement to the game's movement mechanics. I'm also glad to hear that you have added music and sounds, as it can greatly enhance the overall experience. The new menu design looks sleek and modern, and I'm excited to see what the new tower with AOE damage will look like. It's great to see that you have a clear plan for future developments, and I look forward to hearing about the progress of the next tasks. 

Keep up the good work!

Team cybuy

**Comment by MikoHeuken** [comment]
Hello Tower Defense Team,

interesting progress. The menu looks very beautiful and the idea with the sounds are great.
Will you make an animation for the AOE damage, like an expolsion? And will it have another sound than the standart tower?

We look forward to hear more about your project in the future.
Team URH

**Comment by MyMrMush** [comment]
Hello there,

it is very nice to see progress in the project that is not just implementing the requirements from the lecture but effectively bringing the product a step forward. Good work!
Have you considered creating a backdrop for the texts in you main menu? I find it currently a bit hard to read against the vivid background.

Cheers,
Team CapyClue

**Comment by Mohmad-Naser-alnakeshbandi** [comment]
Hello Tower Defense Team,

Well done. The menu looks very beautiful and the idea with the sounds is another good idea from your ideas.
I have a suggestion for the start page of your game, you might add an "🫱" before the words :"play, settings and Quit" so the user could know which word or option is selected 

We look forward to hearing more about your project in the future.
Ghalia Team

**Reply by TimWaeckerle** [reply]
Hi Ghalia Team,

actually, the game is played on an android phone. That means that the "selection" is made by a click with your r finger, so there won't be any time you can select anything. We included a hover effect on each button for a better feeling, but I think that's it. 

Greetings
TD Team

**Comment by 2000eBe** [comment]
Hello Team Tower Defense,

the menu looks great with the fantasy themed background. Do you have any title music to accompany the menu? Where did you source your artwork (also free or maybe someone is very talented in your group?).
It also sound extremely fortunate, that the newest update allows you to work with way more waypoints to create smoother animations - but does it make the process in Unity for you more complicated and how does it affect your performance (if you have any information to that).

Good luck!
Team VIGAD (B6 Course)

**Comment by robin13901** [comment]
Hello TD Team!

It's great to hear about the progress you've made in your Tower Defense game development. The improvements you've implemented sound exciting and will surely enhance the overall gaming experience.

The introduction of new waypoints is a significant enhancement to the movement of enemies in your game. By utilizing the updated functionality from Unity, you've created a smoother and more streamlined path for your enemies to follow. This improvement not only enhances the gameplay but also demonstrates your team's proactive approach in keeping up with the latest advancements and incorporating them into your project.

The addition of music and sounds adds another layer of immersion to the game. It's thoughtful of you to include game settings that allow players to customize their audio preferences. Storing the settings locally in JSON format ensures that players' preferences are remembered every time they start the game.

The updated menu design looks fantastic! It's refreshing to see that you've taken the time to revamp the menu, giving it a fresh and polished appearance. The visual aspect of a game is crucial for attracting players and creating an engaging atmosphere.

The development of new turrets, including one that deals area-of-effect (AOE) damage, shows your commitment to expanding the gameplay mechanics and providing players with diverse strategic options. The combination of both code implementation and design work indicates that you're dedicated to delivering a well-rounded and visually appealing tower selection.

Your plans for future tasks, such as tower upgrades, additional maps, and more towers or upgrades outside of rounds, demonstrate a forward-thinking approach to game development. These additions will undoubtedly contribute to the depth and replay value of your game.

We're excited to see your progress in implementing these upcoming tasks, and we appreciate your commitment to providing regular updates. Keep up the excellent work, TD Team, and we look forward to hearing more about your game in the future.

Best regards,
TuDu
